web/store/data/chats.ts

```diff
@@ -56,7 +56,7 @@ export async function createChat(characterId: string, props: NewChat) {
   const { chat, msg } = createNewChat(char, props)
 
   local.saveChats(chats.concat(chat))
-  local.saveMessages(chat._id, [msg])
+  if (props.greeting) local.saveMessages(chat._id, [msg])
```

Comment:
This brings the localstorage persistence mode in line with what the backend is actually doing (it does not add a greeting if none is provided in the `createChat` payload)